lib/libshare/smb.c

```diff
@@ -64,6 +64,21 @@ static boolean_t smb_available(void);
 
 static sa_fstype_t *smb_fstype;
 
+static void smb_strncpy(char *dst, char *src, size_t strlength)
+{
+	if (src == NULL || dst == NULL)
+		return;
+
+	strncpy(dst, src, strlength);
+	dst[strlength - 1] = '\0';
+}
+
+static void smb_free(char *str)
+{
+	if (str != NULL)
+		free(str);
+}
```

Comment:
This isn't needed.  It's safe to call `free()` with a NULL according to `free(3)`.